### PR TITLE
Use original display name for fields in uploaded tables

### PR DESCRIPTION
**Pull request by crisptrutski**
### Description

It's not a great experience when using foreign languages to have obfuscated field names for CSV uploads.

This change makes sure we preserve the pre-munged name in the display name.

### CODE REVIEW

File: src/metabase/upload.clj [hunk 1]
```diff
@@ -68,6 +70,29 @@
   (let [column-limit (some-> driver driver/column-name-length-limit)]
     (min-safe column-limit (max-bytes :model/Field :name))))
 
+(defn truncate-utf8-bytes
```
Comment (crisptrutski) [inline src/metabase/upload.clj:73]: This is complex and a bit slow, but we should almost never need to execute the no-op branch. Perhaps there's something built in that I just couldn't find.

**Review by sansolovyov — APPROVED** [review]
I'm not an expert in that part of the code base, but changes seem rational.

File: test/metabase/upload_test.clj [hunk 2]
```diff
@@ -1925,6 +1931,28 @@
...
+                   file     (csv-file-with csv-rows)]
+               (testing "The new row is inserted with the values correctly reordered"
+                 (is (= {:row-count 1} (update-csv! action {:file file, :table-id (:id table)})))
+                 (is (= ["name" "α"]
+                        (rest (column-display-names-for-table table))))
+                 (is (= (set (updated-contents action
+                                               [["Obi-Wan Kenobi" nil]]
+                                               [["Everything" "omega"]]))
+                        (set (rows-for-table table)))))
+               (io/delete-file file)))))))))
```
Comment (tsmacdonald) [inline test/metabase/upload_test.clj:1954]: Should probably be  in a `finally` so that it runs if the test fails, etc.

**Review by tsmacdonald — APPROVED** [review]
That imperative `let` in `update-with-csv!` gets thornier and thornier, but that's not really your fault. Looks good to me

**Pull request merged**